debian/rules

```diff
@@ -15,9 +15,16 @@ ifneq (,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
     MAKEFLAGS += -j$(NUMJOBS)
 endif
 
+ifneq (,$(filter BUILD_TYPE=%,$(SCONS_ARGS)))
+    BUILD_TYPE = $(patsubst build_type=%,%,$(filter BUILD_TYPE=%,$(SCONS_ARGS)))
+else
+    BUILD_TYPE = release
+endif
+
 SCONS = scons -j $(NUMJOBS)
 DEB_SCONS_OPTIONS := --config=force --no-rpath USE_INSTALLED=all \
-                     CONF_DIR=$(sysconfdir)/daos PREFIX=$(prefix)
+                     CONF_DIR=$(sysconfdir)/daos PREFIX=$(prefix) \
+                     $(BUILD_TYPE)
```

Comment:
If I understand this correctly, these are the arguments passed to the `scons` command and in the RPM spec file we do this with `BUILD_TYPE=release` by defualt, so don't we need to include `BUILD_TYPE=` here as well?
```suggestion
                     BUILD_TYPE=$(BUILD_TYPE)
```

Reply:
Making the followign change to rules:
No default for BUILD_TYPE; if it is not specified then SConstruct will handle the default

DEB_SCONS_OPTIONS := --config=force --no-rpath USE_INSTALLED=all \
                     CONF_DIR=$(sysconfdir)/daos PREFIX=$(prefix) \
                     $(SCONS_ARGS)

Reply:
> Making the followign change to rules:
> No default for BUILD_TYPE; if it is not specified then SConstruct will handle the default

Which should be confirmed as "secure by default" and compiles out FI if not specified, and open a ticket to get that fixed if it's not the case.
 
> DEB_SCONS_OPTIONS := --config=force --no-rpath USE_INSTALLED=all
> CONF_DIR=$(sysconfdir)/daos PREFIX=$(prefix)
> $(SCONS_ARGS)

As long as `$(SCONS_ARGS)` is empty for the no-FI build.